modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/TypeSafeValueResolverWrapper.java

```diff
@@ -70,6 +71,8 @@ public void initialise() throws InitialisationException {
     if (!valueResolverDelegate.isDynamic() && !InputStream.class.isAssignableFrom(expectedType)) {
       resolver = new CachedResolver(resolver);
     }
+
+    initialiseIfNeeded(valueResolverDelegate, true, muleContext);
```

Comment:
Should we be worried about adding this here?

Reply:
Is this because the RegistryLookupValueResolverWrapper is not getting initialized/injected ? Maybe you can do it when instantiating it.

Reply:
I think that the `initialiseIfNeeded` does not take into account if it was already initialized.